Fix market dropdownmenu default text

saraheisa opened the pull request:
Add text attribute to b-dropdown in filterDropdown component

[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #11423
### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)



[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID:

saraheisa committed:
Fix market dropdownmenu default text

Add text attribute to b-dropdown in filterDropdown component

saraheisa commented:
solving issue #11423

paglias commented on website/client/src/components/ui/filterDropdown.vue:
```diff
@@ -1,7 +1,9 @@
     <span class="dropdown-label">{{ label }}</span>
-    <b-dropdown right="right">
+    <b-dropdown
+    :text="(selectedItem.id.includes('sortBy')) ? selectedItem.id.substring(6) : selectedItem.id"
```
this doesn't seem to be using the correct string version but simply the key meaning it won't be translated correctly, seeing this instead of the correct string that would be the capitalized "Warrior"
<img width="190" alt="Schermata 2019-10-31 alle 12 17 58" src="https://user-images.githubusercontent.com/589911/67942659-9d2d2500-fbd8-11e9-995d-f87d1f2b69ff.png">

I mean that the default selection should match the item listed in the dropdown (with the same text and icon)

saraheisa replied:
I'm new to vue 
So would you please help me with this
How can I grab the value of the dropdown menu

I found out that "text" attribute returns the same value in the dropdown but it only works for class

paglias replied:
I think this is something that might have broken with a recent version of bootstrap vue because the slot is there so it might be a breaking change we missed. @negue any idea? The slot exists here https://github.com/HabitRPG/habitica/pull/11481/files#diff-b7d94aa4b827758be8c8eddcd201c4e4R7 but is not rendered

negue replied:
@prosarahgamal already found the correct slot name :tada:

@paglias yeah another vue-bootstrap upgrade issue I'll go through other components and see if we still have some removed properties / bindings

paglias commented on website/client/src/components/ui/filterDropdown.vue:
```diff
@@ -1,7 +1,9 @@
+    :text="(selectedItem.id.includes('sortBy')) ? selectedItem.id.substring(6) : selectedItem.id"
+    right="right">
       <span
         slot="text"
```
@prosarahgamal I would try to change `slot="text"` with `slot="button-content"` here and remove your other changes

saraheisa replied:
@paglias Thanks a lot for your valuable comments
It works fine now

saraheisa committed:
Fix market dropdownmenu default text 

delete the text attribute form b-dropdown because it wasn't the right  solution
In filterDropdown b-dropdown tag change the span slot to button-content
to match the recent changes in bootstrap vue

paglias commented:
thanks @prosarahgamal, if you have an Habitica UUID or username I'll be happy to award you a contribution tier

saraheisa commented:
> thanks @prosarahgamal, if you have an Habitica UUID or username I'll be happy to award you a contribution tier

Yeah sure!
my username @dev_sarah (ba216907-332d-461a-8e41-c1c5d295e4de)

paglias commented:
@prosarahgamal awarding your contributor tier :)

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!
